src/contexts/CheckContext.js

```diff
@@ -15,11 +15,31 @@ import { irodsVersionComparator } from '../utils'
 
 const validateChecks = (checks) => {
     return checks.reduce((acc, curr) => {
-        if (!curr.name || typeof curr.name !== 'string' || !curr.description || typeof curr.description !== 'string' || !curr.interval_in_seconds || typeof curr.interval_in_seconds !== 'number' || curr.interval_in_seconds < 0 || !curr.checker || (curr.minimum_server_version && curr.maximum_server_version && irodsVersionComparator(curr.maximum_server_version,curr.minimum_server_version) < 0) || typeof curr.active !== 'boolean') {
-            curr.isValid = false
-            curr.active = false
+        let state = {
+            isValid: true,
         }
-        else curr.isValid = true
+ 
+
+        let assert_checkfile_precondition = (state, precondition) => {
+            if (!precondition) {
+                console.log("setting isValid to false", curr.name)
+               state.isValid = false
+            }
+        }
+
+        assert_checkfile_precondition(state, curr.name && typeof curr.name === 'string');
+        
+        assert_checkfile_precondition(state, curr.description && typeof curr.description === 'string');
+        assert_checkfile_precondition(state, curr.interval_in_seconds && typeof curr.interval_in_seconds === 'number' && curr.interval_in_seconds > 0);
+        assert_checkfile_precondition(state, curr.checker && typeof curr.checker === 'function');
+        assert_checkfile_precondition(state, !(curr.minimum_server_version) || typeof curr.minimum_server_version === 'string'); // either min version is not in the file, or it is a string
+        assert_checkfile_precondition(state, !(curr.maximum_server_version) || typeof curr.maximum_server_version === 'string');
+        assert_checkfile_precondition(state, !(curr.minimum_server_version) || !(curr.maximum_server_version) || irodsVersionComparator(curr.maximum_server_version,curr.minimum_server_version) >= 0); // if both min and max version are in the file, then max version must be greater than or equal to min version
+        assert_checkfile_precondition(state, !('active' in curr) || typeof curr.active === 'boolean');
```

Comment:
i like the comments - please put one above each assert_checkfile_precondition rather than at the end of the line.

let em breathe :)